### PR TITLE
Handle term parent on update / create term

### LINKED ISSUE #1204: Support changing parent on hierarchical taxonomies.
On hierarchical taxonomies, a client should be able to update the `parent` of a term. We'll need to make sure the supplied `term_taxonomy_id` is transformed to the `term_id` stored in the database.

### PR DESCRIPTION
**Pull request by joehoyle**
Fixes #1204

### CODE REVIEW

File: lib/endpoints/class-wp-rest-terms-controller.php [hunk 1]
```diff
@@ -193,6 +192,20 @@ public function create_item( $request ) {
 			$args['slug'] = $request['slug'];
 		}
 
+		if ( isset( $request['parent'] ) ) {
+			if ( ! is_taxonomy_hierarchical( $this->taxonomy ) ) {
+				return new WP_Error( 'rest_taxonomy_not_hierarchical', __( "Can not set term parent, taxonomy is not hierarchical." ), array( 'status' => 400 ) );
+			}
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-rest-terms-controller.php]: `parent` isn't present in the schema, so we don't need to worry about the taxonomy check.

File: lib/endpoints/class-wp-rest-terms-controller.php [hunk 2]
```diff
@@ -224,6 +237,20 @@ public function update_item( $request ) {
 			$prepared_args['slug'] = $request['slug'];
 		}
 
+		if ( isset( $request['parent'] ) ) {
+			if ( ! is_taxonomy_hierarchical( $this->taxonomy ) ) {
+				return new WP_Error( 'rest_taxonomy_not_hierarchical', __( "Can not set term parent, taxonomy is not hierarchical." ), array( 'status' => 400 ) );
+			}
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-rest-terms-controller.php]: Same as above.

File: lib/endpoints/class-wp-rest-terms-controller.php (hunk 1 above)
Comment (joehoyle) [inline lib/endpoints/class-wp-rest-terms-controller.php]: This is here to provide feedback to someone sending the parent and not realizing. I think we shouldn't proesume everyone is reading the schema

Comment (danielbachhuber) [inline lib/endpoints/class-wp-rest-terms-controller.php]: With #1223, the argument won't be included.

Comment (joehoyle) [inline lib/endpoints/class-wp-rest-terms-controller.php]: As #1223 hasn't been decided o nor included, I'm going to leave this in.

**Pull request merged**